src/components/progress-report/workflow/progress-report-workflow.service.ts

```diff
@@ -100,6 +126,84 @@ export class ProgressReportWorkflowService {
       this.repo.changeStatus(reportId, transition.to),
     ]);
 
-    // TODO(transition.notify);
+    const notificationInfo = await this.repo.getNotificationInfoByReportId(
+      reportId,
+    );
+    const projectId = notificationInfo[0]?.projectId ?? undefined;
+    const languageDisplayName = notificationInfo[0]?.languageDisplayName;
+    const reportQtrLabel = `Q${fiscalQuarter(
+      notificationInfo[0]?.start,
+    )} FY${fiscalYear(notificationInfo[0]?.start)}`;
+
+    let emailList: string[] = [];
+    notificationInfo.forEach((item) => {
+      item.role?.forEach((role) => {
+        if (transition.notify.TeamMembersWithRoles.includes(role)) {
+          if (item.email) emailList.push(item.email);
+        }
+      });
+    });
+
+    transition.notify.Emails.forEach((element: string) => {
+      if (element.includes('@')) {
+        emailList.push(element);
+      } else if (element === 'MONDAY_EMAIL_DISTRO') {
+        emailList.push(this.configService.mondayEmailDistro);
+      }
+    });
+
+    emailList = uniq(emailList);
+    const recipients = emailList.map((email) =>
+      this.prepareRecipientObject(email),
+    );
+
+    const changedBy = await this.userService.readOne(session.userId, session);
+    const project = await this.projectService.readOne(projectId, session);
+
+    const notifications: EmailReportStatusNotification[] = [];
+    recipients.forEach((recipient) =>
+      notifications.push({
```

Comment:
There's a lot of "loop over X, map, and push result to Y" here. Instead these could just be `.map` calls and skip mutating an existing variable.

```suggestion
    const notifications = recipients.map(
      (recipient): EmailReportStatusNotification => ({
```